### PR TITLE
[BEAM-5668] Revert "[BEAM-5442] Pass unknown SDK pipeline options to Runner (#6557)"

### PR DESCRIPTION
**Pull request by swegner**
This reverts commit db1e6a1c79e44b7026e78d4f0df168234a13eed7.

This change broke post-commits: https://issues.apache.org/jira/browse/BEAM-5668

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by swegner** [comment]
Run Python PostCommit

**Comment by swegner** [comment]
Not sure yet whether #6557 is the culprit, but if so we should rollback and then re-submit with the fix.  [Apache Beam Postcommit Policies](https://beam.apache.org/contribute/postcommits-policies-details/index.html#rollback_first),

**Comment by swegner** [comment]
CC @mxm @tweise @robertwb

**Comment by mxm** [comment]
Ah actually just validates runner and not the entire postcommit. Will make sure next time to run it.

File: sdks/python/apache_beam/options/pipeline_options.py [hunk 1]
```diff
@@ -214,19 +213,7 @@ def get_all_options(self, drop_default=False):
...
     for cls in subset.values():
       cls._add_argparse_args(parser)  # pylint: disable=protected-access
-    known_args, unknown_args = parser.parse_known_args(self._flags)
-    # Parse args which are not known at this point but might be recognized
-    # at a later point in time, i.e. by the actual Runner.
-    if unknown_args and unknown_args[0] != '':
-      logging.info("Parsing unknown args: %s", unknown_args)
-      for arg in unknown_args:
-        if arg.startswith('--'):
-          parser.add_argument(arg.split('=', 1)[0], nargs='?')
```
Comment (aaltay) [inline sdks/python/apache_beam/options/pipeline_options.py]: I believe you need to change here to. Argparse accepts arguments without an equals sign as well. For example `--myargname=myargvalue` would be equivalent to `--myargname myargvalue`. In the latter case, the `split` will return the whole thing.

**Comment by mxm** [comment]
Thanks @swegner for spotting this so quickly. We have fixed PostCommit via #6589. Will address the root cause in a follow-up.

**Pull request closed without merging**